Tests/Integration/CloudToDeviceMessageTests.cs

```diff
@@ -1059,7 +991,13 @@ public async Task When_Takes_Too_Long_Getting_FcntDown_Should_Abandon_Message()
 
             var payload = simulatedDevice.CreateUnconfirmedDataUpMessage("1234", fcnt: PayloadFcnt);
             var rxpk = payload.SerializeUplink(simulatedDevice.AppSKey, simulatedDevice.NwkSKey).Rxpk[0];
-            using var request = CreateWaitableRequest(rxpk, useRealTimer: true);
+            using var request = WaitableLoRaRequest.Create(rxpk, new[]
```

Comment:
As much as I like the idea, this does create a hard dependency on the calls to determine the elapsed time. If we add one more call to validate the elapsed time, we will have to change all the tests using this. We have only one at the moment, but eventually, we may want to have more (I know there are more tests trying to emulate exactly this). Not sure a helper would really make things better to have a single method where we can identify them?

Reply:
My thinking was that the only way to truly enforce this would be to make the `WaitableLoRaRequest.Create` that takes an enumerable private and introduce different methods/configurations on `WaitableLoRaRequest`. I decided that with only one test it was not worth it, but given that we can already anticipate that more tests will reuse this exact configuration. I went ahead and implemented it with 9165ee7ea701761457affad4c05195639bed31c2

Reply:
nice. much better than what I had in mind.